aggregation/src/main/java/fk/prof/aggregation/MethodIdLookup.java

```diff
@@ -0,0 +1,49 @@
+package fk.prof.aggregation;
+
+import com.koloboke.collect.map.hash.HashLongObjMaps;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class MethodIdLookup {
+  //-2 is reserved for placeholder global root method of all stack-trace
+  public final static long GLOBAL_ROOT_METHOD_ID = -2l;
+  public final static String GLOBAL_ROOT_METHOD_SIGNATURE = "~ ROOT ~.()";
+  //-1 is reserved for placeholder unclassifiable roor method of all incomplete stack-traces
+  public final static long UNCLASSIFIABLE_ROOT_METHOD_ID = -1l;
+  public final static String UNCLASSIFIABLE_ROOT_METHOD_SIGNATURE = "~ UNCLASSIFIABLE ~.()";
+
+  //Counter to generate method ids in auto increment fashion
+  //0 is reserved. No method is assigned 0. Counter starts with 1 during assignment
+  private AtomicLong counter = new AtomicLong(0);
```

Comment:
method_id as returned by recorder is long. That gets mapped to a different method id in the instance of this class. Since the source was long, it prompted me to keep the remapped method Id, long as well.